MetaMask/metamask-extension
# Updating tokensChainsCache[chainId].data to object from array in user state

Part of #16057 
## Explanation
In token detection v1, the cache data stored in tokensChainsCache[chainId].data was an array but in Token detection V2, we have updated that to an object. While upgrading to the new version, the array gets restructured into an object resulting in integer keys starting from 0. 

This migration will update the user state to the objects with the token addresses as keys.


## Screenshots/Screencaps


### Before
<img width="362" alt="Screenshot 2022-11-16 at 2 17 04 PM" src="https://user-images.githubusercontent.com/43930900/202273166-1eb5fb0e-2ed0-4942-bd02-cc3fe058691b.png">


### After
<img width="355" alt="Screenshot 2022-11-16 at 2 19 15 PM" src="https://user-images.githubusercontent.com/43930900/202273850-ed0a580e-87f4-47af-84fe-09095226d5e2.png">



## Manual Testing Steps
1. Checkout and build any version before v10.18.0
2. Import the wallet and enable token detection.
3. Switch to Mainnet and click `refreshlist` to detect tokens with balance.
4. Switch to develop, build and open the metamask UI -- the tokens will not have the logo and if you hit refreshlist SNX token with 0 balance will be detected.
5. Now switch to this feature branch and build. Open the metamask UI, and the correct logo will appear for the tokens.


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
